openslides_backend/action/action_worker.py

```diff
@@ -0,0 +1,288 @@
+            ram = psutil.virtual_memory()
+            response = {
+                "success": False,
+                "message": f"ram total:{ram.total} available:{ram.available} percent:{ram.percent} used:{ram.used} free.{ram.free}",
+                "action_error_index": 0,
+                "action_data_error_index": 0,
+            }
+            self.datastore.write_action_worker(
+                WriteRequest(
+                    events=[
+                        Event(
+                            type=EventType.Update,
+                            fqid=self.fqid,
+                            fields={
+                                "timestamp": current_time,
+                                "result": response,
+                            },
+                        )
+                    ],
+                    information={self.fqid: ["timestamp during running action_worker"]},
+                    user_id=self.user_id,
+                    locked_fields={},
+                )
+            )
+            self.logger.debug(
+                f"timestamp during running action_worker '{self.fqid} {self.action_names}': {current_time}"
+            )
+
+    def final_action_worker_write(self, action_worker_thread: "ActionWorker") -> None:
+        current_time = round(time())
+        with self.datastore.get_database_context():
+            state = "end"
+            if hasattr(action_worker_thread, "exception"):
+                response = {
+                    "success": False,
+                    "message": str(action_worker_thread.exception),
+                    "action_error_index": 0,
+                    "action_data_error_index": 0,
+                }
+            elif hasattr(action_worker_thread, "response"):
+                response = action_worker_thread.response
+            else:
+                state = "aborted"
+                response = {
+                    "success": False,
+                    "message": "action_worker aborted without any specific message",
+                    "action_error_index": 0,
+                    "action_data_error_index": 0,
+                }
+            self.datastore.write_action_worker(
+                WriteRequest(
+                    events=[
+                        Event(
+                            type=EventType.Update,
+                            fqid=self.fqid,
+                            fields={
+                                "state": state,
+                                "timestamp": current_time,
+                                "result": response,
+                            },
+                        )
+                    ],
+                    information={self.fqid: ["finish action_worker"]},
+                    user_id=self.user_id,
+                    locked_fields={},
+                )
+            )
+            self.logger.debug(
+                f"action_worker '{self.fqid} {self.action_names}': {current_time}"
```

Comment:
Again, the current time does not really make sense, but I would expect some wording of "final" or "finished".

Reply:
done